libsolidity/formal/SMTChecker.cpp

```diff
@@ -533,29 +529,32 @@ void SMTChecker::inlineFunctionCall(FunctionCall const& _funCall)
 	}
 }
 
+void SMTChecker::abstractFunctionCall(FunctionCall const& _funCall)
+{
+	vector<smt::Expression> smtArguments;
+	for (auto const& arg: _funCall.arguments())
+		smtArguments.push_back(expr(*arg));
+	defineExpr(_funCall, (*m_expressions.at(&_funCall.expression()))(smtArguments));
+	m_uninterpretedTerms.push_back(&_funCall);
+	setSymbolicUnknownValue(expr(_funCall), _funCall.annotation().type, *m_interface);
```

Comment:
Does this still allow to prove `if (a == b) assert(blockhash(a) == blockhash(b));`?

Reply:
It only restricts the type, right?

Reply:
Yes, that's only the numerical range from the type.